### ISSUE TITLE
Custom resources  other than AWS/Custom cause a cfn_nag exception

### ISSUE THREAD

**Issue by twellspring** [stelligent/cfn-model#52]
Custom resources with a type other than `AWS` or `Custom` cause a cfn_nag exception.  
Update /lib/cfn-model/parser/cfn_parser.rb to allow other types of custom resources.

Example template
```
Resources:
  VPC:
    Type: Versent::Network::VPC
[...]
Transform:
  Name: "123456789012::VPC"
```

Error
```
$ cfn_nag --debug ./aws/*.yaml
/usr/local/bundle/gems/cfn-model-0.4.0/lib/cfn-model/parser/cfn_parser.rb:218:in `generate_resource_class_from_type': Unknown namespace in resource type: Versent (RuntimeError)
	from /usr/local/bundle/gems/cfn-model-0.4.0/lib/cfn-model/parser/cfn_parser.rb:189:in `rescue in class_from_type_name'
```

Reference: 
- https://github.com/stelligent/cfn_nag/issues/223
- https://github.com/stelligent/cfn_nag/issues/153

**Comment by jeshan** [comment]
My scenario is about the same. I've boiled it down the following.

does not work (notice the dashes in `Type`):
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  Res:
    Type: Custom::x-y-z
```


works:
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  Res:
    Type: Custom::xyz
```

**Comment by phelewski** [comment]
This cfn_nag issue is also triggering this issue:
- https://github.com/stelligent/cfn_nag/issues/322

**Comment by erickascic** [comment]
cfn-model 0.4.9 fixes this and cfn-nag 0.4.52 uses cfn-model 0.4.9

**Comment by erickascic** [comment]
I fixed the hyphen issue too.  Custom::x-y-z will create a class with the name Xyz
